File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        Stack<Character> stack = new Stack<>();
        
        for(int i=0;i<s.length();i++){
            if(stack.isEmpty()){
                if(s.charAt(i)==')'){
                    return false;
                }
                stack.add(s.charAt(i));
            }else if(s.charAt(i) == ')' && stack.peek() =='('){
                stack.pop();
            }else{
                stack.add(s.charAt(i));
            }
        }
        
        if(stack.size()>0){
            return false;
        }else{
            return true;
        }
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    stack = []
    
    for b in s:  
        if b == '(':
            stack.append(b)
        else:
            if len(stack) == 0:
                answer = False
                break
            elif stack[-1] == '(':
                stack.pop()
            else:
                answer = False
                break
    
    if len(stack) > 0 :
        answer = False
    
    return answer